dao/secret_dao_test.go

```diff
@@ -0,0 +1,95 @@
+package dao
+
+import (
+	"testing"
+
+	"github.com/RedHatInsights/sources-api-go/internal/testutils"
+	"github.com/RedHatInsights/sources-api-go/util"
+)
+
+func TestSecretListUserOwnership(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+	testutils.SkipIfNotSecretStoreDatabase(t)
+	schema := "user_ownership"
+	SwitchSchema(schema)
+
+	userIDWithOwnRecords := "user_based_user"
+	otherUserIDWithOwnRecords := "other_user"
+	userIDWithoutOwnRecords := "another_user"
+
+	userWithOwnRecords, err := CreateUserForUserID(userIDWithOwnRecords, tenantId)
+	if err != nil {
+		t.Error(err)
+	}
+
+	secret1, err := CreateSecretByName("Secret 1", &tenantId, &userWithOwnRecords.Id)
+	if err != nil {
+		t.Error(err)
+	}
+
+	otherUserWithOwnRecords, err := CreateUserForUserID(otherUserIDWithOwnRecords, tenantId)
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = CreateSecretByName("Secret 2", &tenantId, &otherUserWithOwnRecords.Id)
+	if err != nil {
+		t.Error(err)
+	}
+
+	secretWithoutOwnership, err := CreateSecretByName("Secret 3", &tenantId, nil)
+	if err != nil {
+		t.Error(err)
+	}
+
+	/*
+	  Test 1 - User can see own records and records without ownership
+	*/
+	requestParams := &RequestParams{TenantID: &userWithOwnRecords.TenantID, UserID: &userWithOwnRecords.Id}
+	secretDaoWithUser := GetSecretDao(requestParams)
+
+	secrets, _, err := secretDaoWithUser.List(100, 0, []util.Filter{})
+	if err != nil {
+		t.Errorf(`unexpected error when listing the application authentications: %s`, err)
+	}
```

Comment:
here should probably be `authentications` or `secrets`
in 
`unexpected error when listing the ...: %s`